csharp/extractor/Semmle.Extraction.CSharp/Analyser.cs

```diff
@@ -430,29 +431,74 @@ public void Dispose()
         public int TotalErrors => CompilationErrors + ExtractorErrors;
 
         /// <summary>
-        /// Logs detailed information about this invocation,
-        /// in the event that errors were detected.
+        /// Logs information about the extractor.
         /// </summary>
-        /// <param name="roslynArgs">The arguments passed to Roslyn.</param>
-        public void LogDiagnostics(string[] roslynArgs)
+        public void LogExtractorInfo(string extractorVersion)
         {
             Logger.Log(Severity.Info, "  Extractor: {0}", Environment.GetCommandLineArgs().First());
-            if (extractor != null)
-                Logger.Log(Severity.Info, "  Extractor version: {0}", extractor.Version);
-
+            Logger.Log(Severity.Info, "  Extractor version: {0}", extractorVersion);
             Logger.Log(Severity.Info, "  Current working directory: {0}", Directory.GetCurrentDirectory());
+        }
+
+        /// <summary>
+        /// Logs information about the extractor, as well as the arguments to Roslyn.
+        /// </summary>
+        /// <param name="roslynArgs">The arguments passed to Roslyn.</param>
+        /// <returns>A Boolean indicating whether to proceed with extraction.</returns>
+        public bool LogRoslynArgs(string[] roslynArgs, string extractorVersion)
+        {
+            LogExtractorInfo(extractorVersion);
+            Logger.Log(Severity.Info, $"  Arguments to Roslyn: {string.Join(' ', roslynArgs)}");
 
-            if (roslynArgs != null)
+            var csharpLogDir = Extractor.GetCSharpLogDirectory();
+            var tempFile = Path.Combine(csharpLogDir, $"csharp.{Path.GetRandomFileName()}.txt");
+
+            bool argsWritten;
+            using (var streamWriter = new StreamWriter(new FileStream(tempFile, FileMode.Append, FileAccess.Write)))
             {
-                Logger.Log(Severity.Info, $"  Arguments to Roslyn: {string.Join(' ', roslynArgs)}");
+                streamWriter.WriteLine($"Arguments to Roslyn: {string.Join(' ', roslynArgs)}");
```

Comment:
I don't think we want this, because we want to be able to @-include the arguments file on the command line arguments for debugging purposes, if we need to reproduce extraction of a particular compilation command.

Reply:
The reason it is there is because we need the hash of the file to include all arguments to Roslyn, not just the ones from the response file, in order to identify identical compilations. However, it appears that response files can contain comments, so adding this line as a comment should work.

Reply:
The comment worked beautifully, nice find. However, the hash of the file and the contents of the file could be different. You hash all the "inputs" (i.e. the command line), and could even have an empty file just to indicate that the arguments have been seen.